crates/matrix-sdk-common/src/deserialized_responses.rs

```diff
@@ -688,10 +702,9 @@ pub enum UnableToDecryptReason {
     MalformedEncryptedEvent,
 
     /// Decryption failed because we're missing the megolm session that was used
-    /// to encrypt the event.
-    ///
-    /// TODO: support withheld codes?
-    MissingMegolmSession,
+    /// to encrypt the event. If the key was withheld on purpose the
+    /// associated code is attached, None means no withheld code was received.
+    MissingMegolmSession(Option<WithheldCode>),
```

Comment:
I'd be somewhat in favour of making this a struct variant rather than a tuple variant: I think it would be clearer, and allow for easier extension in the future:

```suggestion
    MissingMegolmSession { withheld_code: Option<WithheldCode> },
```

However, it's not a strong opinion.